fdbserver/DDTxnProcessor.actor.cpp

```diff
@@ -929,9 +981,19 @@ void DDMockTxnProcessor::rawFinishMovement(MoveKeysParams& params,
 		mgs->allServers.at(id).setShardStatus(params.keys, MockShardStatus::COMPLETED, mgs->restrictSize);
 	}
 
+	// remove destination servers from source servers
 	ASSERT_EQ(srcTeams.size(), 0);
 	for (auto& id : srcTeams.front().servers) {
-		mgs->allServers.at(id).removeShard(params.keys);
+		if (!std::binary_search(params.destinationTeam.begin(), params.destinationTeam.end(), id)) {
```

Comment:
I think there should be a comment somewhere here explaining why we can make the assumption that `params.destinationTeam` is sorted.